src/Stratis.Bitcoin.IntegrationTests/Miners/ProofOfStakeMintCoinsSteps.cs

```diff
@@ -0,0 +1,177 @@
+
+        private const string PosWallet = "poswallet";
+        private const string PosWalletAccount = "account 0";
+        private const string PosWalletPassword = "password";
+
+        public ProofOfStakeMintCoinsSpecification(ITestOutputHelper outputHelper) : base(outputHelper)
+        {
+        }
+
+        protected override void BeforeTest()
+        {
+            this.sharedSteps = new SharedSteps();
+            this.nodeGroupBuilder = new NodeGroupBuilder();
+        }
+
+        protected override void AfterTest()
+        {
+            this.nodeGroupBuilder.Dispose();
+        }
+
+        private void a_proof_of_work_node_with_wallet()
+        {
+            this.nodes = this.nodeGroupBuilder
+                    .CreateStratisPowMiningNode(PowMiner)
+                    .MineCoinsFast()
+                    .Start()
+                    .NotInIBD()
+                    .WithWallet(PowWallet, PowWalletPassword)
+                    .Build();
+
+            this.powSenderAddress = this.nodes[PowMiner].FullNode.WalletManager().GetUnusedAddress(new WalletAccountReference(PowWallet, PowWalletAccount));
+            var wallet = this.nodes[PowMiner].FullNode.WalletManager().GetWalletByName(PowWallet);
+            this.powSenderPrivateKey = wallet.GetExtendedPrivateKeyForAddress(PowWalletPassword, this.powSenderAddress).PrivateKey;
+        }
+
+        private void it_mines_genesis_and_premine_blocks()
+        {
+            this.sharedSteps.MinePremineBlocks(this.nodes[PowMiner], PowWallet, PowWalletAccount, PowWalletPassword);
+        }
+
+        private void mine_coins_to_maturity()
+        {
+            this.nodes[PowMiner].GenerateStratisWithMiner(100);
+            this.sharedSteps.WaitForBlockStoreToSync(this.nodes[PowMiner]);
+        }
+
+        private void a_proof_of_stake_node_with_wallet()
+        {
+            this.nodeGroupBuilder
+                    .CreateStratisPosNode(PosStaker)
+                    .MineCoinsFast()
+                    .Start()
+                    .NotInIBD()
+                    .WithWallet(PosWallet, PosWalletPassword)
+                    .Build();
+        }
+
+        private void it_syncs_with_proof_work_node()
+        {
+            this.posReceiverAddress = this.nodes[PosStaker].FullNode.WalletManager().GetUnusedAddress(new WalletAccountReference(PosWallet, PosWalletAccount));
+            var wallet = this.nodes[PosStaker].FullNode.WalletManager().GetWalletByName(PosWallet);
+            this.posReceiverPrivateKey = wallet.GetExtendedPrivateKeyForAddress(PosWalletPassword, this.posReceiverAddress).PrivateKey;
+
+            this.nodes[PosStaker].SetDummyMinerSecret(new BitcoinSecret(this.posReceiverPrivateKey, this.nodes[PosStaker].FullNode.Network));
+            this.nodeGroupBuilder.WithConnections().Connect(PowMiner, PosStaker);
+        }
+
+        private void create_tx_to_send_million_coins_from_pow_wallet_to_pos_node_wallet()
```

Comment:
Since this step actually sends it too, perhaps a rename to    `sends_a_million_coins_from_pow_wallet_to_pos_wallet`

Reply:
Renamed 👍